features/nanostack/FEATURE_NANOSTACK/mbed-mesh-api/source/mesh_system.c

```diff
@@ -31,7 +31,7 @@
 
 /* Heap for NanoStack */
 #if !MBED_CONF_MBED_MESH_API_USE_MALLOC_FOR_HEAP
-static uint8_t app_stack_heap[MBED_CONF_MBED_MESH_API_HEAP_SIZE + 1];
+static uint8_t app_stack_heap[MBED_CONF_MBED_MESH_API_HEAP_SIZE + 1] __attribute__(( section( "mesh_heap") ));
```

Comment:
What is this change? I believe this section is not defined for all targets

Reply:
thank you, rebase done, removed the commit

Reply:
L476JG memory is splitted in 2 banks with not consecutive addresses. In the original project only 96kb  SRAM1 is used but this memory is not enought for the project. We moved this array to the 32kb SRAM2 bank which the original project don't use. See the STM32L476XX.ld file modified.

Reply:
This is target specific then, should not be in this code, you should be able to place object files in the sections (might be better place in the linker to do this than in this generic file).

@SeppoTakalo please review this

Reply:
I changed the STM32L476XX.ld file with: 

    .mesh_heap (COPY):
    {
        *mesh_system.o
    } > SRAM2

and removed *__attribute__(( section( "mesh_heap") ));*
The bin file is built without errors and the .map files seems good: [mbed-os-sensor-node.map.txt](https://github.com/ARMmbed/mbed-os/files/1321753/mbed-os-sensor-node.map.txt)

but when I compile this mbed-os version for the *mbed-os-example-client* the demo doesn't run:

  *Starting mbed Client example
  [EasyConnect] IPv6 mode
  [EasyConnect] Using Mesh
  [EasyConnect] Connecting to Mesh..
  [EasyConnect] MAC address
  [EasyConnect] Connection to Network Failed -3012!
  Connection to Network Failed - exiting application...*

Please can you help me?

Reply:
cc @JanneKiiskila

Reply:
@SeppoTakalo might be able to help out a bit more, but well - more logs would definitely be needed. Easy-connect itself does nothing there, it's just abstracting the network stack differences into one function we call from the app side.

@SeppoTakalo - we have one of those boards available here in Oulu.

Reply:
Please note that there is some funny thing on-going with the MAC address. In all normal scenarios the MAC address printing should work - however, in case - there is no MAC? Is the radio driver working?

Reply:
**Please note that there is some funny thing on-going with the MAC address.**
Yes it's true. In the mbed_app.json I have: "spirit1.mac-address": "{0x23, 0x23, 0x23, 0x23, 0x23, 0x23, 0x23, 0x23}"

This with trace enabled:

*Starting mbed Client example
[EasyConnect] IPv6 mode
[EasyConnect] Using Mesh
[EasyConnect] Connecting to Mesh..
[DBG ][SPIRIT]: rf_register (728)
[DBG ][SPIRIT]: rf_ack_loop (608)
[DBG ][SPIRIT]: rf_register (775)
[EasyConnect] MAC address
[EasyConnect] Connection to Network Failed -3012!

Connection to Network Failed - exiting application...*

If I comment this (just for test):
/*.mesh_heap (COPY):
{
    *mesh_system.o
} > SRAM2*/
from the .ld file, the radio works:

Starting mbed Client example
[EasyConnect] IPv6 mode
[EasyConnect] Using Mesh
[EasyConnect] Connecting to Mesh..
[DBG ][core]: Allocate Root Tasklet
[DBG ][6lo ]: P.Init

[DBG ][core]: NS Root task Init
[DBG ][sck ]: Socket Tasklet Generated
[DBG ][SPIRIT]: rf_register (728)
[DBG ][SPIRIT]: rf_ack_loop (608)
[DBG ][SPIRIT]: rf_interface_state_control (466)
[DBG ][SPIRIT]: rf_register (775)
[DBG ][SPIRIT]: get_mac_address (797)
[DBG ][SPIRIT]: rf_get_mac_address (559), adr0=23, adr1=23, adr2=23, adr3=23, adr4=23, adr5=23, adr6=23, adr7=23
[DBG ][SPIRIT]: rf_set_mac_address (547), adr0=23, adr1=23, adr2=23, adr3=23, adr4=23, adr5=23, adr6=23, adr7=23
[DBG ][SPIRIT]: rf_set_mac_address (547), adr0=23, adr1=23, adr2=23, adr3=23, adr4=23, adr5=23, adr6=23, adr7=23
[DBG ][nslp]: connect()
[DBG ][mle ]: MLE service init size 32
[DBG ][sck ]: Socket id 0 allocated
[DBG ][m6LND]: Using PSK security mode.
[DBG ][m6LND]: Channel: 1
[DBG ][m6LND]: Channel page: 2
[DBG ][m6LND]: Channel mask: 2
[DBG ][mMIB]: Set key 1
[INFO][addr]: Address added to IF 1: fe80::2123:2323:2323:2323

But without my original modification the client won't connect to the mbed connector due to memory overflow problems.

Reply:
I believe we all agree this change should not be here, and we need to identify the root cause of this problem.

@JanneKiiskila  you noted that you have these boards in Oulu, or @MarceloSalazar in Cambridge to reproduce the problems they are having?